src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+            editor.InsertBefore(oldMethodDeclaration, newMethodDeclaration);
+        }
+
+        public static void UpdateExpressionInOriginalFunction(SemanticDocument semanticDocument,
+            TExpressionSyntax expression, SyntaxNode scope, string parameterName, SyntaxEditor editor,
+            bool allOccurrences, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(semanticDocument.Document);
+            var matches = FindMatches(semanticDocument, expression, semanticDocument, scope, allOccurrences, cancellationToken);
+            var parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(parameterName);
+
+            // Parenthesize the variable, and go and replace anything we find with it.
+            // NOTE: we do not want elastic trivia as we want to just replace the existing code 
+            // as is, while preserving the trivia there.
+            var replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false);
+
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            foreach (var match in matches)
+            {
+                if (match.Equals(expression))
+                {
+                    if (ShouldRemoveVariableDeclaratorContainingExpression(semanticDocument, expression, out var varDecl))
+                    {
+                        var localDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsLocalDeclarationStatement(node));
+                        if (localDeclaration is not null)
+                        {
+                            editor.RemoveNode(localDeclaration);
+                        }
+                    }
+                    else
+                    {
+                        parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(generator.Identifier(parameterName).WithAdditionalAnnotations(RenameAnnotation.Create()));
+                        replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false);
+                        editor.ReplaceNode(match, replacement);
+                    }
+                }
+                else
+                {
+                    editor.ReplaceNode(match, replacement);
+                }
+            }
+        }
+
+        private static int GetInsertionIndex(SemanticDocument document, Compilation compilation,
+            ISyntaxFactsService syntaxFacts, SyntaxNode methodDeclaration, CancellationToken cancellationToken)
+        {
+            var parameterList = syntaxFacts.GetParameterList(methodDeclaration);
+            Contract.ThrowIfNull(parameterList);
+            var symbol = document.SemanticModel.GetDeclaredSymbol(parameterList.GetRequiredParent(), cancellationToken);
+            Contract.ThrowIfNull(symbol);
+            var parameterSymbolList = ((IMethodSymbol)symbol).Parameters;
```

Comment:
```suggestion
            var symbol = document.SemanticModel.GetDeclaredSymbol(parameterList.GetRequiredParent(), cancellationToken) as IMethodSymbol method;
            Contract.ThrowIfNull(symbol);
            var parameterSymbolList = method.Parameters;
```

HOwever, same comment still applies.  I"ve seen you compute this over and over in different ways. just compute thisall the way up front and pass it around.  it's clearly very important state for the purposes of this entire operation.

Reply:
removed and passed in